### DISCUSSION TITLE
Class "Fideloper\Proxy\TrustProxies" not found

### DISCUSSION THREAD

**Discussion by stephenmeehanuk** [statamic/cms#5591] [Troubleshooting] [answered]
Hi

Upgraded from `3.3.38` to `3.3.2` and I'm seeing this error?

<img width="1398" alt="Screenshot 2022-03-22 at 11 01 54" src="https://user-images.githubusercontent.com/72254741/159467580-6e3e76f4-8061-43a2-af99-f12bbb56c419.png">

I ran `composer update` 

This is my composer.json

```
"require": {
        "php": "^8.0.2",
        "fruitcake/laravel-cors": "^2.0.5",
        "guzzlehttp/guzzle": "^7.2",
        "laravel/framework": "^9.2",
        "laravel/horizon": "^5.7",
        "laravel/tinker": "^2.7",
        "predis/predis": "^1.1",
        "simonhamp/statamic-heroicons": "^1.0",
        "statamic/cms": "3.3.*",
        "laravel/sanctum": "^2.14"
    },
    "require-dev": {
        "barryvdh/laravel-debugbar": "^3.6",
        "spatie/laravel-ignition": "^1.0",
        "fakerphp/faker": "^1.9.1",
        "mockery/mockery": "^1.4.4",
        "nunomaduro/collision": "^6.1",
        "phpunit/phpunit": "^9.5.10",
        "laravel/sail": "^1.13"
    },
```   

I found [this](https://stackoverflow.com/questions/47193349/class-fideloper-proxy-trustproxies-not-found-laravel) on Stackoverflow, looks like it was missing `fideloper/proxy`

I tried to install: `composer require fideloper/proxy` but got this error?

```
 Error 

  Undefined constant Illuminate\Http\Request::HEADER_X_FORWARDED_ALL

  at vendor/fideloper/proxy/config/trustedproxy.php:48
     44▕      * - 'HEADER_X_FORWARDED_AWS_ELB' (If you are using AWS Elastic Load Balancer)
     45▕      *
     46▕      * @link https://symfony.com/doc/current/deployment/proxies.html
     47▕      */
  ➜  48▕     'headers' => Illuminate\Http\Request::HEADER_X_FORWARDED_ALL,
     49▕ 
     50▕ ];
     51▕ 

      +9 vendor frames 
  10  [internal]:0
      Illuminate\Foundation\Application::Illuminate\Foundation\{closure}(Object(Fideloper\Proxy\TrustedProxyServiceProvider))

      +5 vendor frames 
  16  artisan:37
      Illuminate\Foundation\Console\Kernel::handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
Script @php artisan package:discover --ansi handling the post-autoload-dump event returned with error code 1

Installation failed, reverting ./composer.json and ./composer.lock to their original content.
```

Anyone else come across this?

**Comment by duncanmcclean** [comment, marked as answer]
Replace the contents of this file (`app/Http/Middleware/TrustProxies.php`) with the contents of this one: https://github.com/statamic/statamic/blob/master/app/Http/Middleware/TrustProxies.php

You should be good to go after that 👍

**Reply by stephenmeehanuk** [reply]
Hi,

Thanks for the reply, I replaced the content of `TrustProxies.php` ... Unfortunately there's a new error.

```
Undefined constant Illuminate\Http\Request::HEADER_X_FORWARDED_ALL
```

![Screenshot 2022-03-22 at 15 06 39](https://user-images.githubusercontent.com/72254741/159513912-807dc1d3-3e27-438b-a38a-a5255d17e7ae.png)

Is there anything else I can try?

**Reply by duncanmcclean** [reply]
Hmm, that should work fine as long as you're on Laravel 9. Not sure why it's not working 🤔

**Reply by stephenmeehanuk** [reply]
When I first ran this update, there was an issue with my internet. The installation failed halfway, I ran `composer update` again this morning thinking the install process would start over. I guess it didn't

I ran `composer require laravel/framework`, it's fixed this issue. Phew.

Thanks for your help.

**Reply by duncanmcclean** [reply]
No worries - glad you managed to get it sorted!
